code/modules/projectiles/guns/energy/special.dm

```diff
@@ -757,5 +757,110 @@
 	M.mimic_type = mimic_type
 	..()
 
+/obj/item/gun/energy/detective
+	name = "Detective's energy revolver"
+	desc = "A 'modern' take on the classic projectile revolver. Shift-click to clear active tracked target or clear linked pinpointer."
+	icon_state = "det_placeholder"
+	modifystate = TRUE
+	ammo_type = list(/obj/item/ammo_casing/energy/detective, /obj/item/ammo_casing/energy/detective/tracker_warrant)
+	ammo_x_offset = 4
+	/// If true, this gun is tracking something and cannot track another mob
+	var/tracking_target_UID
+	/// Used to track if the gun is overcharged
+	var/overcharged
+	/// Yes, this gun has a radio, welcome to 2022
+	var/obj/item/radio/headset/det_gun/Announcer
+	/// Used to link back to the pinpointer
+	var/linked_pinpointer_UID
+
+/obj/item/radio/headset/det_gun
+	requires_tcomms = FALSE
+
+/obj/item/gun/energy/detective/Initialize(mapload, ...)
+	. = ..()
+	Announcer = new /obj/item/radio/headset/det_gun(src)
+	Announcer.config(list("Security" = 1))
+
+/obj/item/gun/energy/detective/ShiftClick(mob/user)
```

Comment:
add an examine override giving players info about the shift click functionality.